include/preprocess/crhmc/analytic_center.h

```diff
@@ -0,0 +1,178 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2020 Vissarion Fisikopoulos
+// Copyright (c) 2018-2020 Apostolos Chalkis
+// Copyright (c) 2022-2022 Ioannis Iakovidis
+
+// Contributed and/or modified by Ioannis Iakovidis, as part of Google Summer of
+// Code 2022 program.
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+// References
+// Yunbum Kook, Yin Tat Lee, Ruoqi Shen, Santosh S. Vempala. "Sampling with
+// Riemannian Hamiltonian
+// Monte Carlo in a Constrained Space"
+#ifndef ANALYTIC_CENTER_H
+#define ANALYTIC_CENTER_H
+#include "Eigen/Eigen"
+#include "PackedCSparse/PackedChol.h"
+#include "preprocess/crhmc/crhmc_utils.h"
+#include "preprocess/crhmc/opts.h"
+#include <fstream>
+#include <iostream>
+#include <vector>
+#ifndef SIMD_LEN
+#define SIMD_LEN 0
+#endif
+const size_t chol_k2 = (SIMD_LEN == 0) ? 1 : SIMD_LEN;
+
+using NT = double;
+using MT = Eigen::Matrix<NT, Eigen::Dynamic, Eigen::Dynamic>;
+using VT = Eigen::Matrix<NT, Eigen::Dynamic, 1>;
+using SpMat = Eigen::SparseMatrix<NT>;
+using CholObj = PackedChol<chol_k2, int>;
+using Triple = Eigen::Triplet<double>;
+using Tx = FloatArray<double, chol_k2>;
+using Opts = opts<NT>;
+/*This function computes the analytic center of the polytope*/
+//And detects additional constraint that need to be added
+// x - It outputs the minimizer of min f(x) subjects to {Ax=b}
+// C - detected constraint matrix
+//     If the domain ({Ax=b} intersect dom(f)) is not full dimensional in {Ax=b}
+//     because of the dom(f), the algorithm will detect the collapsed dimension
+//     and output the detected constraint C x = d
+// d - detected constraint vector
+template <typename Polytope>
+std::tuple<VT, SpMat, VT> analytic_center(SpMat const &A, VT const &b, Polytope &f, Opts const &options, VT x = VT::Zero(0, 1))
+{
+  // initial conditions
+  int n = A.cols();
+  int m = A.rows();
+  if (x.rows() == 0 || !f.barrier.feasible(x))
+  {
+    x = f.barrier.center;
+  }
+
+  VT lambda = VT::Zero(n, 1);
+  int fullStep = 0;
+  NT tConst = 0;
+  NT primalErr = std::numeric_limits<NT>::infinity();
```

Comment:
what happens if `std::numeric_limits<NT>::has_infinity == false`? Why not using `std::numeric_limits<NT>::max()`?

Reply:
You are right I changed `+/-std::numeric_limits<NT>::infinity()` with `std::numeric_limits<NT>::max()` and `std::numeric_limits<NT>::lowest()`.